pkg/operator/k8sutil/service.go

```diff
@@ -92,3 +100,103 @@ func ParseServiceType(serviceString string) v1.ServiceType {
 	}
 	return v1.ServiceType("")
 }
+
+func IsServiceExported(ctx context.Context, c *clusterd.Context, service *v1.Service) (bool, error) {
+	client, err := mcsv1Client.NewForConfig(c.KubeConfig)
+	if err != nil {
+		return false, errors.Wrap(err, "failed to get mcs-api client")
+	}
+
+	_, err = client.MulticlusterV1alpha1().ServiceExports(service.Namespace).Get(ctx, service.Name, metav1.GetOptions{})
+	if err != nil {
+		if kerrors.IsNotFound(err) {
+			return false, nil
+		}
+		return false, errors.Wrapf(err, "failed to get exported service %q", service.Name)
+	}
+
+	return true, nil
+}
+
+// ExportService exports the service using MCS API and returns the external IP of the exported service
+func ExportService(ctx context.Context, c *clusterd.Context, service *v1.Service) (string, error) {
+	client, err := mcsv1Client.NewForConfig(c.KubeConfig)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to get mcs-api client")
+	}
+
+	serviceExport := &mcsv1a1.ServiceExport{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            service.Name,
+			Namespace:       service.Namespace,
+			OwnerReferences: service.GetOwnerReferences(),
+		},
+	}
+
+	_, err = client.MulticlusterV1alpha1().ServiceExports(service.Namespace).Create(ctx, serviceExport, metav1.CreateOptions{})
+	if err != nil && !kerrors.IsAlreadyExists(err) {
+		return "", errors.Wrapf(err, "failed to create exported service %q", service.Name)
+	}
+
+	var exportedIP string
+	var serviceExportError error
+	exportedIP, err = getExportedServiceIP(fmt.Sprintf("%s.%s.svc.clusterset.local", service.Name, service.Namespace))
+
+	if err != nil {
+		serviceExportError = errors.Wrapf(err, "failed to get exported service IP for %q", service.Name)
+		err := verifyExportedService(ctx, client, service.Name, service.Namespace)
+		if err != nil {
+			serviceExportError = errors.Wrapf(err, "failed to create service export successfully for the service %q", service.Name)
+		}
+		return "", serviceExportError
+	}
+
+	return exportedIP, nil
+}
+
+// verifies if the ServiceExport status conditions to determine if the service was exported correctly
+func verifyExportedService(ctx context.Context, client *mcsv1Client.Clientset, name, namespace string) error {
+	exportedService, err := client.MulticlusterV1alpha1().ServiceExports(namespace).Get(ctx, name, metav1.GetOptions{})
+	if err != nil {
+		return errors.Wrapf(err, "failed to get exported service %q", name)
+	}
+
+	if len(exportedService.Status.Conditions) == 0 {
+		return fmt.Errorf("no status conditions available in the exported service %q", name)
+	}
+
+	for _, condition := range exportedService.Status.Conditions {
+		if condition.Type == mcsv1a1.ServiceExportValid && condition.Status == corev1.ConditionFalse {
+			return fmt.Errorf(*condition.Message)
+		}
+	}
+
+	return nil
+}
+
+func getExportedServiceIP(fqdn string) (string, error) {
+	retryCount := 5
+	serviceExportWaitTime := 10 * time.Second
```

Comment:
How about we retry slightly faster, and retry for up to 100s? I don't have a good reason other than 50s seems a bit short in general to retry with K8s.
```suggestion
	retryCount := 20
	serviceExportWaitTime := 5 * time.Second
```

Reply:
updated.